include/seqan3/std/concepts

```diff
@@ -17,6 +17,66 @@
 
 #ifdef __cpp_lib_concepts        // C++20 ranges available
 #include <concepts>
+
+namespace std
+{
+inline namespace deprecated
+{
+/*!\brief This should be named default_initializable.
+ * \deprecated Use std::default_initializable instead.
+ */
+template<class T>
+concept default_constructible = default_initializable<T>;
+
+/*!\brief This should be removed.
+ * \deprecated No viable alternative.
+ */
+template <class B>
+concept boolean =
+    movable<std::remove_cvref_t<B>> &&
+    requires(std::remove_reference_t<B> const & b1,
+             std::remove_reference_t<B> const & b2, bool const a)
+    {
+        requires convertible_to<const std::remove_reference_t<B>&, bool>;
+        !b1;      requires convertible_to<decltype(!b1), bool>;
+        b1 && a;  requires same_as<decltype(b1 && a), bool>;
+        b1 || a;  requires same_as<decltype(b1 || a), bool>;
+        b1 && b2; requires same_as<decltype(b1 && b2), bool>;
+        a && b2;  requires same_as<decltype(a && b2), bool>;
+        b1 || b2; requires same_as<decltype(b1 || b2), bool>;
+        a || b2;  requires same_as<decltype(a || b2), bool>;
+        b1 == b2; requires convertible_to<decltype(b1 == b2), bool>;
+        b1 == a;  requires convertible_to<decltype(b1 == a), bool>;
+        a == b2;  requires convertible_to<decltype(a == b2), bool>;
+        b1 != b2; requires convertible_to<decltype(b1 != b2), bool>;
+        b1 != a;  requires convertible_to<decltype(b1 != a), bool>;
+        a != b2;  requires convertible_to<decltype(a != b2), bool>;
+    };
+} // namespace std::deprecated
+} // namespace std
+
+namespace std::detail
+{
+/*!\interface   std::detail::weakly_equality_comparable_with <>
+ * \tparam t1   The first type to compare.
+ * \tparam t2   The second type to compare.
+ * \brief       Requires the two operands to be comparable with `==` and `!=` in both directions.
+ * \sa          https://en.cppreference.com/w/cpp/concepts/weakly_equality_comparable_with
+ */
+//!\cond
+template <class T, class U>
+concept weakly_equality_comparable_with =
```

Comment:
Why the `weakly`? Cannot we use just `equality_comparable_with`, which is more intuitive? Otherwise, please say in the documentation what weakly means.

Reply:
This is the point; we use it internally even though we might use the non-exposition only version.

Reply:
But this is not the scope of this PR; this PR just enables to able to compile it with `-std=c++2a`.

Reply:
ok!